### PR TITLE
fix: add sleep rule before start

### PR DESCRIPTION
**Pull request by mathiasbio**
### This PR:

**BREAK GLASS IN CASE OF KEY_ERROR EMERGENCY**

This PR is an ugly solution to the key_error issue (https://github.com/Clinical-Genomics/BALSAMIC/issues/1300) we sometimes see when a rule finishes before all the jobs have been submitted. It simply adds a 5 minute sleep rule before the actual workflow starts.

If we see this error appearing during verification and validation we can merge this. 

image

After 5 minutes downstream rules begin:
image

Added: 
- Sleep 300 seconds before starting analysis

### Tests run:
- [x] TGA T+N case unitedbeagle is running successfully
- [ ] TGA T+N case unitedbeagle finished successfully

### Review and tests: 
- [ ] Tests pass
- [ ] Code review
- [ ] New code is executed and covered by tests, and test approve

### CODE REVIEW

File: CHANGELOG.rst [hunk 1]
```diff
@@ -34,6 +34,8 @@ Added:
 * CNV report for TGA workflow https://github.com/Clinical-Genomics/BALSAMIC/pull/1339
 * `wkhtmltopdf` to system requirements https://github.com/Clinical-Genomics/BALSAMIC/pull/1339
 * Store WGS CNV report plots https://github.com/Clinical-Genomics/BALSAMIC/pull/1347
+* Sleep rule before start to fix key_error https://github.com/Clinical-Genomics/BALSAMIC/pull/1310
+
```
Comment (ivadym) [inline CHANGELOG.rst:38]: Should be under [13.0.1]

File: BALSAMIC/workflows/PON.smk [hunk 2]
```diff
@@ -48,6 +49,9 @@ bam_dir: str = Path(result_dir, "bam", "").as_posix() + "/"
 cnv_dir: str = Path(result_dir, "cnv", "").as_posix() + "/"
 qc_dir: str = Path(result_dir, "qc", "").as_posix() + "/"
 
+# Pre run parameters
+seconds_before_start: int = SLEEP_BEFORE_START
```
Comment (ivadym) [inline BALSAMIC/workflows/PON.smk:53]: Couldn't we just use directly `SLEEP_BEFORE_START` in the rules after importing it in the smks?

File: BALSAMIC/snakemake_rules/misc/sleep.rule [hunk 3]
```diff
@@ -0,0 +1,15 @@
+
+rule sleep_before_start:
+    """Wait 120s before starting any processing to avoid key_error issue."""
```
Comment (ivadym) [inline BALSAMIC/snakemake_rules/misc/sleep.rule:3]:
```suggestion
    """Wait 300s before starting any processing to avoid key_error issue."""
```

**Review by ivadym — APPROVED** [review]
👍

File: BALSAMIC/workflows/PON.smk (hunk 2 above)
Comment (mathiasbio) [reply BALSAMIC/workflows/PON.smk]: Yes you're right of course : ) removed. Let me run a case to confirm it's still working

File: CHANGELOG.rst [hunk 4]
```diff
@@ -1,3 +1,11 @@
+[13.0.1]
+-------
+
+Added:
+^^^^^^
+* Sleep rule before start to fix key_error https://github.com/Clinical-Genomics/BALSAMIC/pull/1310
```
Comment (ivadym) [inline CHANGELOG.rst:6]:
```suggestion
* Sleep rule before start to fix key_error https://github.com/Clinical-Genomics/BALSAMIC/pull/1311
```

**Pull request merged**